core/sql/regress/executor/TEST002

```diff
@@ -1102,7 +1102,7 @@ SELECT student_name,
 
 SELECT student_name,
           GROUP_CONCAT(DISTINCT test_score
-                    ORDER BY test_score SEPARATOR '-')
+                    ORDER BY test_score desc SEPARATOR '-')
```

Comment:
Previous ORDER BY do nothing, it is ignored. So there will be regression failure , since the order is random. 
I can insert more rows into the table, above syntax is the most complex syntax for this feature. But I can design a more complex query to use join for example?

ORDER BY in an aggregate function is regarded as extraOrder requirement, that is the basic idea of this change. Because function is ItemExpr, so it cannot have its own sort requirement as I understand.
Let me first check the shortcut groupby and partial groupby issue